### DISCUSSION TITLE
Doctrine Bigint => String => declare strict_types=1

### DISCUSSION THREAD

**Discussion by ghost** [symfony/symfony#43905] [Q&A] [answered]
Hi everyone,

I am faced with a small problem, I ask for a little light.
When I build the database, I need bigint for handling very large cipher. When I do a mapping:import the annotations are impeccable, when I do the make:entity --regenerate, all bigint columns become strings due to their length. In the doc doctrine, moreover, he points it out.
The concern is, when I set the declare strict_types=1, all the interaction functions with doctrine are necessarily put in red in my IDE, because I send int, in the entity, the functions expect the string.

Is there a solution for this?
Can I do in null | string | int expectations or should I not use declare?

Looking forward to hearing from you,
Cordially,

**Comment by mbabker** [comment, marked as answer]
By design, the DBAL treats bigint values as a string.  See https://www.doctrine-project.org/projects/doctrine-dbal/en/latest/reference/types.html#bigint for details.

You can use the getter and setter in your entity to handle string-to-int (and reverse) transformations so your application can use those bigint values as integers.  You just need to make sure your application is always deployed on 64-bit architecture to avoid the issue described in the docs regarding why bigints are handled as strings.

**Reply by ghost** [reply]
OK,
So as I thought if I do a strict declare, I need to add string | int in the methods.
Thank you
